flang/lib/Optimizer/Transforms/SimplifyFIROperations.cpp

```diff
@@ -122,6 +124,59 @@ mlir::LogicalResult BoxTotalElementsConversion::matchAndRewrite(
   return mlir::failure();
 }
 
+class DoConcurrentConversion
```

Comment:
Let me know if this is not a good place to add that pattern. Looking at all other passes, this seemed the most fitting and I thought it does not make much sense to have a pass just for that.